build/source/engine/read_force.f90

```diff
@@ -20,23 +20,51 @@
 
 module read_force_module
 
+! data types
+USE nrtype                                    ! variable types, etc.
+
+! derived data types
+USE data_types,only:gru_hru_double            ! x%gru(:)%hru(:)%var(:)     (dp)
+
+! constants
+USE multiconst,only:secprday                  ! number of seconds in a day
+
 ! access missing values
-USE globalData,only:realMissing       ! real missing value
-USE globalData,only:integerMissing    ! integer missing value
+USE globalData,only:realMissing               ! real missing value
+USE globalData,only:integerMissing            ! integer missing value
 
 ! access the mapping betweeen GRUs and HRUs
-USE globalData,only:gru_struc         ! gru-hru mapping structures
+USE globalData,only:gru_struc                 ! gru-hru mapping structures
 
 ! access the minimum and maximum HRUs in the file
 USE globalData,only:ixHRUfile_min,ixHRUfile_max
 
-! define data types
-USE data_types,only:gru_hru_double    ! x%gru(:)%hru(:)%var(:)     (dp)
+! global data on the forcing file
+USE globalData,only:data_step                 ! length of the data step (s)
+USE globalData,only:forcFileInfo              ! forcing file info
+USE globalData,only:dJulianStart              ! julian day of start time of simulation
+USE globalData,only:refJulday                 ! reference time (fractional julian days)
+USE globalData,only:refJulday_data            ! reference time for data files (fractional julian days)
+USE globalData,only:fracJulDay                ! fractional julian days since the start of year
+USE globalData,only:yearLength                ! number of days in the current year
+USE globalData,only:nHRUfile                  ! number of days in the data file
 
+! global metadata
+USE globalData,only:time_meta,forc_meta       ! metadata structures
+USE var_lookup,only:iLookTIME,iLookFORCE      ! named variables to define structure elements
+
+! file paths
+USE summaFileManager,only:INPUT_PATH          ! path of the forcing data file
+
+! privacy
 implicit none
 private
 public::read_force
 
+! global parameters
+real(dp),parameter  :: verySmall=1e-3_dp      ! tiny number
+real(dp),parameter  :: smallOffset=1.e-8_dp   ! small offset (units=days) to force ih=0 at the start of the day
```

Comment:
Should these be defined globally?

Reply:
These parameters are specific to `read_force.f90`. We have similar parameters in other routines (set to different values) which are also local in scope.